graphene_sqlalchemy/types.py

```diff
@@ -294,7 +324,10 @@ def get_query(cls, info):
         return get_query(model, info.context)
 
     @classmethod
-    def get_node(cls, info, id):
+    async def get_node(cls, info, id):
```

Comment:
But isn't `execute_sync` calling `graphql_impl` in https://github.com/graphql-python/graphql-core/blob/4a63f97bd8dc676a984bf81787a62cab7848d2ae/src/graphql/graphql.py#L130, which in turn calls `execute` in https://github.com/graphql-python/graphql-core/blob/4a63f97bd8dc676a984bf81787a62cab7848d2ae/src/graphql/graphql.py#L185, which eventually calls execute_fields containing the code you referred to in https://github.com/graphql-python/graphql-core/blob/4a63f97bd8dc676a984bf81787a62cab7848d2ae/src/graphql/execution/execute.py#L439-L456? 🙂

Reply:
I've asked in the graphql-python slack about this, maybe we get more clarification there.

Reply:
I played a bit around with it and I think I understand the issue better and it seems like you are right 😉 
If running `schema.execute` instead of `await schema.execute_async` I end up getting an error thrown from here: https://github.com/graphql-python/graphene-sqlalchemy/blob/a47dbb369c5e7a057affe7cb98fa15d3acf29cd8/graphene_sqlalchemy/types.py#L288 🤔 
I will try to update the tests so they run both `execute` and `execute_async` which should ensure that the other cases are handled too, right? 🙂

Reply:
I added tests for the synchronous executes of get_node back in. I don't really like the code that I added for `resolve_connection_async`, but unluckily the mixture of async and sync made the creation of the connection object hard 🙈 
Currently mixing a synchronous execution with an async session yields a not straight forward error that essentially states that what was returned by the downstream function is a coroutine not an SQLAlchemy model. Adding something similar to https://github.com/graphql-python/graphql-core/blob/4a63f97bd8dc676a984bf81787a62cab7848d2ae/src/graphql/graphql.py#L144 seems to make sense in this case. Could you give me a heads up on a good position to add it? 🙂